docs/Glossary.md

```diff
@@ -115,6 +115,23 @@ A *location key* is a string that is unique to a particular [`location`](#locati
 
     type LocationListener = (location: Location) => void;
 
+### LocationSpec
```

Comment:
I'm like ~65% on this.

We lose a bit of conciseness by introducing this concept of a "location spec" object, but I think it's worthwhile to not entirely merge this concept into that of a "location", because of the omission of the `action` and `key` fields.

I'd be totally fine with merging this into the concept of a `location` if people feel that would make more sense, though.